### PR TITLE
[WIP] Make atmospherics more expensive

### PR DESCRIPTION
**Pull request by Allfd**
Initial commit of a throw calculation for atmospherics.

Currently this makes it much more difficult for flying items to kill you and should help minimize the "dancing" behavior around breaches.

This comes at the cost of extra CPU.

### CODE REVIEW

File: code/LINDA/LINDA_turf_tile.dm [hunk 1]
```diff
@@ -320,7 +320,26 @@
...
 				H.Weaken(min(pressure_difference / 10, 10))
 			spawn()
-				throw_at(general_direction, pressure_difference / 10, pressure_difference / 200, null, 0, 0, null)
+				var/max_distance = 14 // reduce by one each calculation to prevent infinate loops.
+				var/min_observed_pressure = INFINITY
+				var/turf/target = get_turf(src)
+				if(isunsimulatedturf(target))
+					return 0
+				var/turf/possible_target = get_turf(src)
+				while(!isunsimulatedturf(target) && max_distance)
```
Comment (Crazylemon64) [inline code/LINDA/LINDA_turf_tile.dm]: to make atmospherics as expensive as possible, this should be:
```
while(TRUE)
  max_distance = 8
```

**Comment by taukausanake** [comment]
If flying items making sweet love to your fleshy bits is an issue wouldn't it be reasonable to make the item land at your feet rather than in front of you so it stops humping you with the wind? I would think that once you share the same tile with an item and you don't move then the offending item would move past you then.

Unless this is a different issue completely

**Comment by Allfd** [comment]
They still hit you, just much more gently.  

This is all subject to change, I just wanted to get something up here so I can test the performance impacts of the underlying set of calls before getting into the fine tuning.

**Comment by Aurorablade** [comment]
ehhhh i dunno if we need to nerf atmos at the cost of CPU.

**Comment by FalseIncarnate** [comment]
Based on the title, I thought this was upping power usage of atmos machinery. Simply making it take up more CPU to do less damage seems odd. Not sure if there is really a better way to do it though.

**Comment by Vivalas** [comment]
Well, adding extra checks to a process that is very repetitive tends to increase CPU time, so you choose either a nerf or efficiency.

**Comment by Jountax** [comment]
Something needs to be done about a single piece of glass being able to instantly murder you just because there's a breach on the other end of the hallway, IMO.

**Comment by taukausanake** [comment]
Again, can't just make the item pass you after it hits you? I would imagine it recoils off your body on to the floor in gravity and in zero-G keep going past

**Comment by Jountax** [comment]
Honestly, much as I hate to say it, getting hit with an object knocking you over (for maybe a few seconds?) might be a more realistic fix, since if you're prone they just fly over you anyway iirc.

**Comment by Allfd** [comment]
Closed,

After talking to Tully, the decision has instead been made to partially revert the atmospheric changes.  This will involve removing the throw aspect, and reducing space wind to pre-merge levels, as such this PR is no-longer relevant.

**Pull request closed without merging**